lib/spack/spack/hooks/sbang.py

```diff
@@ -17,9 +17,13 @@
 import spack.store
 
 
-#: Character limit for shebang line.  Using Linux's 127 characters
-#: here, as it is the shortest I could find on a modern OS.
-shebang_limit = 127
+#: Character limit for shebang line: 127 for Linux; 511 for Mac. Mac
+#: generates longer temporary file paths, which can lead to failures when
+#: generating sbang scripts for tests.
```

Comment:
This should mention that the numbers here are actually the OS limits (see https://www.in-ulm.de/~mascheck/various/shebang/).  I'm not sure that the point about temp file paths is relevant in the `sbang` hook module.